### PR TITLE
Getblockfrombody AAA test and function

### PR DESCRIPTION
**Pull request by bashbashbash**
#432

### CODE REVIEW

File: internal/handlers/handlers.go [hunk 1]
```diff
@@ -137,3 +139,20 @@ func HandleGetJSONBlockByHeight(fetcher ifaces.IBlockFetcher) func(w http.Respon
 		io.WriteString(w, string(marshalledBlock))
 	}
 }
+
+// GetBlockFromResponse will convert the body of a reponse handler and return a Block
```
Comment (kstola2) [inline internal/handlers/handlers.go]: converts the body of a response* , I don't know what a response handler is

File: internal/handlers/handlers.go [hunk 2]
```diff
@@ -137,3 +139,20 @@ func HandleGetJSONBlockByHeight(fetcher ifaces.IBlockFetcher) func(w http.Respon
...
+// Note - per the documentation on the response struct, the body is never nil
+// TODO how do we test the response is valid if the body is never nil?
```
Comment (kstola2) [inline internal/handlers/handlers.go]: HTTP uses status codes. Please read up on them. 200 would be returned upon successful block retrieval

- [ ] remove TODO

File: internal/handlers/handlers.go [hunk 3]
```diff
@@ -137,3 +139,20 @@ func HandleGetJSONBlockByHeight(fetcher ifaces.IBlockFetcher) func(w http.Respon
...
+	if r.StatusCode == http.StatusBadRequest {
+		return block.Block{}, errors.New("StatusBadRequest")
+	}
+	var requestBody block.JSONBlock
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(r.Body)
+	json.Unmarshal(buf.Bytes(), &requestBody)
+
+	fmt.Printf("%v", requestBody)
+	bodyBlock, _ := requestBody.Unmarshal()
```
Comment (kstola2) [inline internal/handlers/handlers.go]: - [ ] Handle the error returned from `Unmarshal`

File: internal/handlers/handlers_test.go [hunk 4]
```diff
@@ -414,3 +414,71 @@ func TestGetJSONBlockByHeight(t *testing.T) {
...
+			rr := httptest.NewRecorder()
+
+			// Serve the request
+			handler.ServeHTTP(rr, req)
+
+			// Actual block that is recorded in the response
+			actualBlock, err := GetBlockFromResponse(rr.Result())
+
+			// Assert
+			if !reflect.DeepEqual(test.expectedBlock, actualBlock) {
```
Comment (kstola2) [inline internal/handlers/handlers_test.go]: - [ ] Please use this function for comparisons: https://github.com/SIGBlockchain/project_aurum/blob/master/internal/block/block.go#L147

File: internal/handlers/handlers.go (hunk 1 above)
Comment (bashbashbash) [reply internal/handlers/handlers.go]: - [ ]

**Pull request merged**